upstairs/src/in_memory.rs

```diff
@@ -57,28 +57,29 @@ impl BlockIO for InMemoryBlockIO {
     async fn read(
         &self,
         offset: Block,
-        data: Buffer,
-    ) -> Result<(), CrucibleError> {
+        mut data: Buffer,
+    ) -> Result<Buffer, CrucibleError> {
```

Comment:
Annotate this with `#[must_use]` as well?  I'm not sure whether the compiler will warn you if you do `read(..).await?` without using the `Ok(..)` branch.

Reply:
@mkeeter `Result` is already `#[must_use]`, so adding `#[must_use]` to this function won't do anything (and, i think, will result in a compiler warning).

Reply:
covered by 115510f (I think)